### ISSUE TITLE
Core doesn't emit `sessionCompleted` event when session ends

### ISSUE THREAD

**Issue by 10ko** [garden-io/garden#4171]
### Current Behavior

When a non-sync command (e.g. `garden deploy`) is executed, Cloud will receive a `sessionCompleted` event to signal the command completed successfully.
This doesn't seem to happen when running `garden deploy --dev` and after pressing Ctrl+C to exit the command.

### Expected behavior

The `sessionCompleted` event to be emitted.

### Reproducible example

Happy to pair on this, but you should be able to try to deploy the quickstart example and see a session always active in the Dashboard.

### Your environment

* **OS:** 
* **How I'm running Kubernetes:**  

`garden version`

**Comment by eysi09** [comment]
Just briefly, here are the states (recap from in-office conversation):

1. Cloud has command results => We're good. 

2 No command results AND command is persistent => We can render the accurate statuses because we know whether or not it's running. But we should be storing the `persistent=true|false` value in Cloud, currently we just "guess", based on command name. (Sort of like, `if name === "dev" then persistent ...`) 

3. No result AND command is not persistent => Here we don't know anything. It could be running or it could've been killed. Currently we just say that it's `running` unless it started more than 120 minutes ago (IIRC), in which case we say it's `incomplete`. We can solve this be emitting heart beats. 

So in summary, what we need to do:

1. Add a `persistent` column to the session record in Cloud and include that data when starting the session. 
2. Send heartbeats form Core when it's running (at the session level). 

NOTE: Emitting a `sessionCompleted` event is a no-go. We can not trust that it's sent before the process terminates.

**Comment by edvald** [comment]
Is this still the case?

**Comment by vvagaytsev** [comment]
@edvald I think so. We decided to implement the heartbeat in the early patch releases of 0.13.

**Comment by vvagaytsev** [comment]
@10ko is this still an issue?

**Comment by vvagaytsev** [comment]
This might be fixed by #4787 and another recent 0.13 PRs. Closing this as done.
